### PR TITLE
create tests for svg_to_image

### PR DESCRIPTION
**Pull request by rudigiesler**

### CODE REVIEW

**Comment by zoidyzoidzoid** [comment]
Are we gonna catch the new exceptions in the service to raise useful responses?

**Comment by rudigiesler** [comment]
Things like `invalid SVG` should never happen and indicate an internal error, so those shouldn't go through to the user. Things like `Unsupported file format` we probably want to give to the user.

File: mathml_to_image_service/svg_to_image.py [hunk 1]
```diff
@@ -20,21 +28,33 @@ def to_image(svg_string, image_format, max_size):
...
-        stderr=subprocess.DEVNULL)
-
-    if image_format.upper() == 'GIF':
+    try:
         subprocess.check_call(
-            ['convert', STATIC_DIR % png_filename, STATIC_DIR % filename],
+            ['rsvg-convert', tmp_file_name, '-w', '%s' % (max_size,),
+             '--background-color', 'white', '-o', STATIC_DIR % png_filename],
             stderr=subprocess.DEVNULL)
+    except subprocess.CalledProcessError as e:
```
Comment (zoidyzoidzoid) [inline mathml_to_image_service/svg_to_image.py]: Unused `e`

File: mathml_to_image_service/svg_to_image.py [hunk 2]
```diff
@@ -20,21 +28,33 @@ def to_image(svg_string, image_format, max_size):
...
+        os.remove(tmp_file_name)
+        os.remove(STATIC_DIR % png_filename)
+        raise ImageConvertError('Invalid SVG')
+
+    if image_format.upper() == 'GIF':
```
Comment (zoidyzoidzoid) [inline mathml_to_image_service/svg_to_image.py]: Not that it really makes a huge performance difference, but if we check if `extension == '.gif'` it'll save doing the `.upper()` call twice.

File: mathml_to_image_service/svg_to_image.py [hunk 3]
```diff
@@ -20,21 +28,33 @@ def to_image(svg_string, image_format, max_size):
...
+        try:
+            subprocess.check_call(
+                ['convert', STATIC_DIR % png_filename, STATIC_DIR % filename],
+                stderr=subprocess.DEVNULL)
+        except subprocess.CalledProcessError as e:
```
Comment (zoidyzoidzoid) [inline mathml_to_image_service/svg_to_image.py]: Also unused `e`

**Comment by zoidyzoidzoid** [comment]
We can either check the file format in the view/route, or catch that exception and turn it into a response.

File: mathml_to_image_service/svg_to_image.py (hunk 1 above)
Comment (rudigiesler) [inline mathml_to_image_service/svg_to_image.py]: I was going to reraise the error, but then I decided to make my own. Good catch

**Comment by rudigiesler** [comment]
Should we do that in another PR, since the goal of this PR was to create tests?

**Comment by zoidyzoidzoid** [comment]
Sure, sounds good to me. :+1:

**Pull request merged**